Sources/Transaction/Memos/SenderWithPaymentRequestMemoUtils.swift

```diff
@@ -13,44 +13,52 @@ enum SenderWithPaymentRequestMemoUtils {
         receipientViewPrivateKey: RistrettoPrivate,
         txOutPublicKey: RistrettoPublic
     ) -> Bool {
-        memoData.asMcBuffer { memoDataPtr in
-            senderPublicAddress.withUnsafeCStructPointer { publicAddressPtr in
-                receipientViewPrivateKey.asMcBuffer { receipientViewPrivateKeyPtr in
-                    txOutPublicKey.asMcBuffer { txOutPublicKeyPtr in
-                        var matches = false
-                        let result = withMcError { errorPtr in
-                            mc_memo_sender_with_payment_request_memo_is_valid(
-                                memoDataPtr,
-                                publicAddressPtr,
-                                receipientViewPrivateKeyPtr,
-                                txOutPublicKeyPtr,
-                                &matches,
-                                &errorPtr)
-                        }
-                        switch result {
-                        case .success:
-                            return matches
-                        case .failure(let error):
-                            switch error.errorCode {
-                            case .invalidInput:
-                                // Safety: This condition indicates a programming error and can only
-                                // happen if arguments to mc_memo_sender_with_payment_request_memo_is_valid are
-                                // supplied incorrectly.
-                                logger.warning("error: \(redacting: error)")
-                                return false
-                            default:
-                                // Safety: mc_memo_sender_with_payment_request_memo_is_valid should not throw
-                                // non-documented errors.
-                                logger.warning("Unhandled LibMobileCoin error: \(redacting: error)")
-                                return false
-                            }
+        // using in-line function to keep closure levels lower
+        // swiftlint:disable closure_body_length
+        func ffiCall(
+            memoDataPtr: UnsafePointer<McBuffer>,
+            publicAddressPtr: UnsafePointer<McPublicAddress>
+        ) -> Bool {
+            receipientViewPrivateKey.asMcBuffer { receipientViewPrivateKeyPtr in
+                txOutPublicKey.asMcBuffer { txOutPublicKeyPtr in
+                    var matches = false
+                    let result = withMcError { errorPtr in
+                        mc_memo_sender_with_payment_request_memo_is_valid(
+                            memoDataPtr,
+                            publicAddressPtr,
+                            receipientViewPrivateKeyPtr,
+                            txOutPublicKeyPtr,
+                            &matches,
+                            &errorPtr)
```

Comment:
double-indent function call params:
```suggestion
                                memoDataPtr,
                                publicAddressPtr,
                                receipientViewPrivateKeyPtr,
                                txOutPublicKeyPtr,
                                &matches,
                                &errorPtr)
```

Reply:
single indentation is the general rule